File: tawalasoft_cargoflow/public/js/purchase_invoice.js
frappe.ui.form.on("Purchase Invoice", {
   custom_purchase_invoice_type(frm) {
       frm.toggle_reqd(
           "custom_shipment_no",
           frm.doc.custom_purchase_invoice_type === "Landed Cost Invoice"
       );
   },

   refresh(frm) {
       frm.toggle_reqd(
           "custom_shipment_no",
           frm.doc.custom_purchase_invoice_type === "Landed Cost Invoice"
       );
   },
});

File: tawalasoft_cargoflow/services/purchase_invoice.py
import frappe


def validate(doc, method=None):
    if doc.custom_purchase_invoice_type == "Landed Cost Invoice" and not doc.custom_shipment_no:
        frappe.throw(
            "Shipment No is mandatory when Purchase Invoice Type is <b>Landed Cost Invoice</b>.",
            title="Missing Shipment No",
        )